### ISSUE TITLE
Unexpected ts2506 error when extending a protected static nested class 

### ISSUE THREAD

**Issue by whzx5byb** [microsoft/TypeScript#50622]
# Bug Report

### 🔎 Search Terms

circular internal nested class

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDyCuAXApgJ2gbwFDV9ADqgPYrAoAm0EiYiAlsNAJIB2La0AvNKJDBgF8A3DjwB6MdAB60VMgC2RAG7JoAIkIlkZZOTXRERaPLABrVXUTQA7kVSmsArFl5RoAYQAWdEJWQAPFBZyGAQUdGw8aAAzIiIACgBKTFEo3GAiFmoYuK4ecDcAoJC4JDQAOlZ2CKco2rxUqhp6RgAjMHRuVxgi5GDQstRKtg5BRyA)

### 💻 Code

```ts
class Outer {
    protected static Inner = class {};
    // ^ remove "protected" to make it work
}

class Child extends Outer {
    foo() {
        const foo = class extends Outer.Inner {}
    }
    
    static bar = class extends Outer.Inner {}
}
```

### 🙁 Actual behavior

Error: '(Anonymous class)' is referenced directly or indirectly in its own base expression.(2506)

### 🙂 Expected behavior

No error.

---
First reported in https://github.com/microsoft/TypeScript/issues/10017#issuecomment-874381600 but it seems not having received much attention.

**Comment by nwrkbiz** [comment]
I also have exact the same problem.
